web/profiles/custom/yalesites_profile/modules/custom/ys_templated_content/src/Managers/TemplateManager.php

```diff
@@ -0,0 +1,209 @@
+        'preview_image' => '',
+      ],
+    ],
+    'profile' => [
+      '' => [
+        'title' => 'Empty',
+        'description' => 'An empty template.',
+        'filename' => '',
+        'preview_image' => '',
+      ],
+      'faculty' => [
+        'title' => 'Faculty',
+        'description' => 'A template for a faculty profile.',
+        'filename' => 'profile__faculty.yml',
+        'preview_image' => '',
+      ],
+      'student' => [
+        'title' => 'Student',
+        'description' => 'A template for a student profile.',
+        'filename' => 'profile__student.yml',
+        'preview_image' => '',
+      ],
+      'staff' => [
+        'title' => 'Staff',
+        'description' => 'A template for a staff profile.',
+        'filename' => 'profile__staff.yml',
+        'preview_image' => '',
+      ],
+    ],
+  ];
+
+  /**
+   * Constructs the controller object.
+   *
+   * @param \Drupal\Core\Extension\ModuleHandler $module_handler
+   *   The module handler.
+   */
+  public function __construct(
+    ModuleHandler $module_handler,
+  ) {
+    $this->moduleHandler = $module_handler;
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public static function create(ContainerInterface $container) {
+    return new static(
+      $container->get('module_handler'),
+      $container->get('ys_templated_content.template_filename_helper'),
+    );
+  }
+
+  /**
+   * Get the template options for the currrent content type.
+   *
+   * @param string $content_type
+   *   The content type.
+   * @param string $template
+   *   The template.
+   *
+   * @return string
+   *   The file path.
+   */
+  public function getFilenameForTemplate($content_type, $template) {
+    $filename = $this->templates[$content_type][$template]['filename'];
+    return $this
+      ->moduleHandler
+      ->getModule('ys_templated_content')
+      ->getPath() . $this::TEMPLATE_PATH . $filename;
+  }
+
+  /**
+   * Get the template options for the currrent content type.
+   *
+   * @return array
+   *   The template options.
+   */
+  public function getCurrentTemplates($content_type) : array {
+    $templates = [];
+    // Return an empty array if there is no content type.
+    if ($content_type) {
+      // For each key and title value, create an array of the key and title.
+      foreach ($this->templates[$content_type] as $key => $template) {
+        $templates[$key] = $template['title'];
+      }
+    }
+    return $templates;
+  }
+
+  /**
+   * Get the template options for the currrent content type.
+   *
+   * @param string $content_type
+   *   The content type.
+   * @param string $template_name
+   *   The template name.
+   *
+   * @return array
+   *   The template options.
+   */
+  public function getTemplateDescription($content_type, $template_name) {
+    return $this->templates[$content_type][$template_name]['description'] ?? "Hi";
+  }
+
+  /**
+   * Get the templates.
+   */
+  public function getTemplates() {
```

Comment:
This is picky so feel free to ignore me. But this function and the `getCurrentTemplates` are similar enough - you could make the `$content_type` argument optional and then if one is passed, juts get the templates with the specified content type, or if none is passed, get all of them.

Reply:
Ah not a bad idea.  I can try that.  :D